internal/controller/apiextensions/composite/composition_render.go

```diff
@@ -83,9 +83,11 @@ func RenderComposedResourceMetadata(cd, xr resource.Object, n ResourceName) erro
 		return errors.Errorf(errFmtNamePrefixLabel, xcrd.LabelKeyNamePrefixForComposed)
 	}
 
-	// We also set generate name in case we haven't yet named this composed
-	// resource.
-	cd.SetGenerateName(xr.GetLabels()[xcrd.LabelKeyNamePrefixForComposed] + "-")
+	// We recommend composed resources let us generate a name for them. They're
+	// allowed to explicitly specify a name if they want though.
+	if cd.GetName() == "" && cd.GetGenerateName() == "" {
```

Comment:
Do we need this fix on main as well?

Reply:
I don't think it's urgent.

It seems like it'll only be an issue when composing resources with opinions about the format of their `metadata.name`, which no MRs have to my knowledge. So it'll be important for main when main starts to support arbitrary resources.